docs/runtimes/functions.md

```diff
@@ -1,27 +1,73 @@
+
+![MLRun Function](../_static/images/mlrun_concepts_architecture.png)
+
+After the MLRun functions and workflows are created and [**registered into the project**](../projects/create-project.html), they are invoked using the project object. 
+This workflow pairs especially well with {ref}`ci-cd-automate`.
+		
+## More about functions	
+
+There are various kinds of MLRun functions with different capabilities, however, 
+there are commonalities across all functions. In general, an MLRun function looks like the following:
+
+<img src="../_static/images/mlrun-functions.png" alt="mlrun-architecture" width="700"/><br>
 
 MLRun supports:
-- Real-time functions for: {ref}`serving <serving-graph>`, APIs, and stream processing (based on the high-performance {ref}`Nuclio <nuclio-real-time-functions>` engine). 
-- Batch functions (based on Kubernetes jobs, {ref}`Spark <spark-operator>`, {ref}`Dask <dask-overview>`, {ref}`Horovod <horovod>`, etc.)
+- Real-time runtimes for: {ref}`serving <serving-graph>`, APIs, and stream processing (based on the high-performance {ref}`Nuclio <nuclio-real-time-functions>` engine). 
+- Batch runtimes (based on Kubernetes jobs, {ref}`Spark Operator<spark-operator>`, {ref}`Dask <dask-overview>`, {ref}`Horovod <horovod>`, etc.)
+
+The [different function runtimes](../concepts/functions-overview.html) take care of automatically transforming the code and spec to fully 
+managed and elastic services over Kubernetes, which saves significant operational overhead, 
+addresses scalability, and reduces infrastructure costs. The function parameters and capabilities are explained in more detail in {ref}`create-and-use-functions`.
 
-Function objects are all inclusive (code, spec, API, and metadata definitions), which allows placing them 
-in a shared and versioned function market place. This means that different members of the team can produce or 
-consume functions. Each function is versioned and stored in the MLRun database with a unique hash code, 
+Function objects are all inclusive, containing the code and all the operational aspects: ({ref}`image <images-usage>`, required packages, 
+{ref}`pod resource congifuration (replicas, CPU/GPU/memory limits, volumes, Spot vs. On-demand nodes, pod priority, node affinity) <configuring-job-resources>`, 
+[storage](../runtimes/function-storage.html), environment, metadata definitions, etc.). Each function is versioned and stored in the MLRun database with a unique hash code, 
 and gets a new hash code upon changes.
 
+When you create a function, you specify its name, which appears in the MLRun UI, as well as the Kubernetes pod.
+
+Depending on the job type, the actual python function 
+can be included in the code as a {py:func}`~mlrun.handler`. You can also use the handler to log results and artifacts, see {ref}`decorators-and-auto-logging`.
```

Comment:
This sentence is also confusing. As the `handler` you point to here is the MLRun decorator. I think a better explanation would be something like:

> MLRun functionality such as logging results and artifacts can be added to existing code using the {py:func}`~mlrun.handler` decorator. You can also use MLRun auto-logging to automatically track machine learning processes such as training a model and record results and artifacts seamlessly while they execute. See {ref}`decorators-and-auto-logging`.

Reply:
Worth to note that the handler decorator is now effective by default since the introduction of packagers.

Reply:
@alonmr  what do you mean "effective by default"? Enabled?

Reply:
https://github.com/mlrun/mlrun/blob/development/mlrun/package/__init__.py#L51

Reply:
updated to 'enabled by default'